packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -939,139 +972,749 @@ describe('NetworkController', () => {
       });
     });
 
-    describe('if a provider has been set', () => {
-      describe('assuming that the version of the current network is different from the network in state', () => {
-        it('updates the network in state to match', async () => {
-          const messenger = buildMessenger();
-          await withController(
-            { messenger, state: { networkId: null } },
-            async ({ controller }) => {
-              await setFakeProvider(controller, {
-                stubs: [
-                  {
-                    request: { method: 'net_version' },
-                    response: { result: '12345' },
+    Object.values(INFURA_NETWORKS).forEach(({ networkType }) => {
```

Comment:
This is the main change made in this PR: the existing tests were adapted to run once per built-in network, and once for custom RPC networks. This is how the extension unit tests work.